### ISSUE TITLE
yolov8-obb 框不全问题

### ISSUE THREAD

**Issue by xiaoyao123456789** [ultralytics/ultralytics#20833]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

我需要训练针对于道路病害检测的obb模型，但是不论怎么调整数据集，也会存在超长尺寸框会框不全的问题。我尝试了增加P6yaml+yolov8/11-obb.pt训练都是这样，且数据集是严丝合缝处理好的，无问题。具体效果如下图

![Image](https://github.com/user-attachments/assets/4219b944-4ec0-4bad-9c21-cc2e5047e92e)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xiaoyao123456789, thank you for reaching out and sharing your experience with Ultralytics 🚀! This is an automated response to help you get the fastest assistance possible. An Ultralytics engineer will review your issue and provide further help soon.

We recommend checking the [Docs](https://docs.ultralytics.com/) for guides, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, which cover many common questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/), including the relevant code, a sample of your data, and any logs to help us debug the problem.

If your question is about custom training or dataset formatting, please share as much detail as possible, such as dataset samples and training logs, and review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community for more support:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for sharing knowledge

## Upgrade

Please make sure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). Update with:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be used in these verified environments (with all dependencies and [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for your report and for helping improve Ultralytics!

**Comment by xiaoyao123456789** [comment]
> 你可以尝试更小的 imgsz

您好，我有尝试640的，但是效果并没有改进。

下图是我使用640尺寸训练40epoch时模型推理效果
![Image](https://github.com/user-attachments/assets/e491de46-166d-4cba-abee-cb3b53d12d1c)
更困惑的是到60epoch时模型推理效果如下，更差了
![Image](https://github.com/user-attachments/assets/2c641093-a64c-4b14-b99e-f15e21ac6605)

我看到了类似朋友的疑问，与我的有些相似，目前我也更改了reg_max=1，但是没理解为啥这位朋友说改成1为关闭dfl loss
https://github.com/ultralytics/ultralytics/issues/8962

**Comment by Y-T-G** [comment]
You can use imgsz 320

**Comment by glenn-jocher** [comment]
The issue you're experiencing with incomplete bounding boxes for very long/large objects is common in OBB detection. Besides trying smaller image sizes like 320, I'd recommend checking your dataset annotation format - ensure your OBB coordinates follow the YOLO OBB format with normalized values between 0 and 1 using the four corner points `class_index x1 y1 x2 y2 x3 y3 x4 y4`. You might also want to experiment with different anchor configurations or consider using multiscale training with the `scale` parameter to help the model better handle objects of varying sizes.

**Comment by glenn-jocher** [comment]
Looking at your continued issues with incomplete bounding boxes in OBB detection, you might also want to check if your extremely long objects are being filtered out during preprocessing. The OBB training pipeline automatically filters out boxes that are too small (`rw >= 2` and `rh >= 2` pixels), but for very elongated objects, even if they're long, one dimension might fall below this threshold. You could try adjusting the `box` loss weight in your hyperparameters or ensuring your dataset contains sufficient examples of these elongated road defects at various scales during training.

**Comment by pearl88** [comment]
@xiaoyao123456789  大佬，问下问题解决了吗？
